### ISSUE TITLE
[GITLAB] Suggestions are generated twice

### ISSUE THREAD

**Issue by pushpanktugnawat** [Codium-ai/pr-agent#353]
I am using pr-agent with GitLab and highly utilizing the review and suggestion command but suggestions are generated twice every time.

Is this a known issue? If yes could you help me to provide how to fix it?

**Comment by mrT23** [comment]
can you share a reproducible example ? does it happen all the time, or once in a while ?

**Comment by pushpanktugnawat** [comment]
It has happened every time in the last few days now. I can share the example image but I don't have any working repo where I can share the same.

**Comment by pushpanktugnawat** [comment]
It has happened every time in the last few days now. I can share the example image but I don't have any working repo where I can share the same.

**Comment by pushpanktugnawat** [comment]
It has happened every time in the last few days now. I can share the example image but I don't have any working repo where I can share the same.

**Comment by mrT23** [comment]
ok, i will try to debug it later today

**Comment by mrT23** [comment]
Thanks for the feedback

https://github.com/Codium-ai/pr-agent/pull/354 should fix it
what happened was that we tried to publish all suggestions together (to be efficient).
but when one suggestion was invalid (and it can happen), the others were still published, yet we also invoked the fallback, so we had duplications

In any case, should work now.
i also worked on the 'review' feature in GitLab, and added links of the 'relevant line'.


Feel free to share more issues you encounter in GitLab.
